File: tests/snapshots/match.py
# Match → Rust's `match`. Each arm is `pattern => body`; the patterns are real
# Rust — literals, ranges (`..=`), alternation (`|`), and the `_` wildcard.

def main():
    score: int = 75
    _m0 = score
    match _m0:
        case 100:
            print('perfect')
        case _ if 90 <= _m0 <= 99:
            print('excellent')
        case _ if 70 <= _m0 <= 89:
            print('good')
        case 0 | 1 | 2:
            print('very low')
        case _:
            print('somewhere in between')


if __name__ == "__main__":
    main()

File: tests/snapshots/match.rs
// Match → Rust's `match`. Each arm is `pattern => body`; the patterns are real
// Rust — literals, ranges (`..=`), alternation (`|`), and the `_` wildcard.

fn vbr_main() -> Result<(), String> {
    let score: i64 = 75;
    match score {
        100 => {
            println!("perfect");
        }
        90 ..= 99 => {
            println!("excellent");
        }
        70 ..= 89 => {
            println!("good");
        }
        0 | 1 | 2 => {
            println!("very low");
        }
        _ => {
            println!("somewhere in between");
        }
    }
    Ok(())
}

fn main() {
    if let Err(error) = vbr_main() {
        eprintln!("Error: {error}");
        std::process::exit(1);
    }
}
